modules/exploits/example_webapp.rb

```diff
@@ -0,0 +1,145 @@
+# a bug in an arbitrary web server
+#
+###
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  #
+  # This exploit affects a webapp, so we need to import HTTP Client
+  # to easily interact with it.
+  #
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        # The Name should be just like the line of a Git commit - software name,
+        # vuln type, class. It needs to fit in 50 chars ideally. Preferably apply
+        # some search optimization so people can actually find the module.
+        # We encourage consistency between module name and file name.
+        'Name'           => 'Sample Webapp Exploit',
+        'Description'    => %q(
+            This exploit module illustrates how a vulnerability could be exploited
+          in a webapp.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => ['h00die'],
+        'References'     =>
+          [
+            [ 'OSVDB', '12345' ],
+            [ 'EDB', '12345' ],
+            [ 'URL', 'http://www.example.com'],
+            [ 'CVE', '1978-1234']
+          ],
+        'Platform'       => ['python'],
+        'Privileged'     => false,
+        'Arch'           => ARCH_PYTHON,
+        'Targets'        =>
+          [
+            [ 'Automatic Target', {}]
+          ],
+        'DisclosureDate' => "Apr 1 2013",
+        # Note that this is by index, rather than name. It's generally easiest
+        # just to put the default at the beginning of the list and skip this
+        # entirely.
+        'DefaultTarget'  => 0
+      )
+    )
+    # set the default port, and a URI that a user can set if the app isn't installed to the root
+    register_options(
+      [
+        Opt::RPORT(80),
+        OptString.new('USERNAME', [ true, 'User to login with', 'admin']),
+        OptString.new('PASSWORD', [ false, 'Password to login with', '123456']),
+        OptString.new('TARGETURI', [ true, 'The URI of the Example Application', '/example/'])
+      ], self.class
+    )
+  end
+
+  #
+  # The sample exploit checks the index page to verify the version number is exploitable
+  # we use a regex for the version number
+  #
+  def check
+    # we want to handle cases where the port/target isn't open/listening gracefully
+    begin
+      # only catch the response if we're going to use it, in this case we do for the version
+      # detection.
+      res = send_request_cgi(
+        'uri'       => normalize_uri(target_uri.path, 'index.php'),
+        'method'    => 'GET'
+      )
+      # gracefully handle if res comes back as nil, since we're not guaranteed a response
+      # also handle if we get an unexpected HTTP response code
+      fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to web service - no response") if res.nil?
+      fail_with(Failure::UnexpectedReply, "#{peer} - Invalid credentials (response code: #{res.code})") if res.code == 200
```

Comment:
is this correct? Throw an invalid credentials error if we get a `200 OK` on a `GET`?

Reply:
nope good catch, the 'logic' of 200 being an issue was post login, not index load.